sdk/src/wallet/account/operations/transaction/high_level/send_amount.rs

```diff
@@ -52,20 +94,80 @@ impl AccountHandle {
     pub async fn prepare_send_amount(
         &self,
         addresses_with_amount: Vec<AddressWithAmount>,
-        options: Option<TransactionOptions>,
+        options: impl Into<Option<TransactionOptions>>,
     ) -> crate::wallet::Result<PreparedTransactionData> {
         log::debug!("[TRANSACTION] prepare_send_amount");
-        let mut outputs = Vec::new();
+        let options = options.into();
+        let rent_structure = self.client.get_rent_structure().await?;
         let token_supply = self.client.get_token_supply().await?;
 
-        for address_with_amount in addresses_with_amount {
-            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address_with_amount.address)?;
+        let account_addresses = self.addresses().await?;
+        let default_return_address = account_addresses.first().ok_or(Error::FailedToGetRemainder)?;
+
+        let local_time = self.client.get_time_checked().await?;
+
+        let mut outputs = Vec::new();
+        for AddressWithAmount {
+            address,
+            amount,
+            return_address,
+            expiration,
+        } in addresses_with_amount
+        {
+            let (address, bech32_hrp) = Address::try_from_bech32_with_hrp(address)?;
+            let return_address = return_address
+                .map(|address| Ok::<_, Error>(Address::try_from_bech32_with_hrp(address)?.0))
```

Comment:
You can simply use `try_from_bech32` if you just need the address

Reply:
Ah, thanks.

Reply:
Any reason to check the hrp like the above receiving does?

Reply:
Yeah actually we should probably check it as well